hhs_oauth_server/tests.py

```diff
@@ -45,3 +49,239 @@ def test_int_values(self):
+            ("1EL4TE5MK74", False),
+            ("1EO4TE5MK74", False),
+            ("1EI4TE5MK74", False),
+            ("1EB4TE5MK74", False),
+            ("1EZ4TE5MK74", False),
+
+            # Position 4 is invalid
+            # Position 4 – numeric values 0 thru 9
+            ("1EGATE5MK74", False),
+
+            # Position 5 is invalid
+            # Position 5 – alphabetic values A thru Z (minus S, L, O, I, B, Z)
+            ("1EG41E5MK74", False),
+            ("1EG4SE5MK74", False),
+            ("1EG4LE5MK74", False),
+            ("1EG4OE5MK74", False),
+            ("1EG4IE5MK74", False),
+            ("1EG4BE5MK74", False),
+            ("1EG4ZE5MK74", False),
+
+            # Position 6 is invalid
+            # Position 6 – alphanumeric values 0 thru 9and A thru Z (minus S, L, O, I, B, Z)
+            ("1EG4TS5MK74", False),
+            ("1EG4TL5MK74", False),
+            ("1EG4TO5MK74", False),
+            ("1EG4TI5MK74", False),
+            ("1EG4TB5MK74", False),
+            ("1EG4TZ5MK74", False),
+
+            # Position 7 is invalid
+            # Position 7 – numeric values 0 thru 9
+            ("1EG4TEAMK74", False),
+
+            # Position 8 is invalid
+            # Position 8 – alphabetic values A thru Z (minus S, L, O, I, B, Z)
+            ("1EG4TE51K74", False),
+            ("1EG4TE5SK74", False),
+            ("1EG4TE5LK74", False),
+            ("1EG4TE5OK74", False),
+            ("1EG4TE5IK74", False),
+            ("1EG4TE5BK74", False),
+            ("1EG4TE5ZK74", False),
+
+            # Position 9 is invalid
+            # Position 9 – alphabetic values A thru Z (minus S, L, O, I, B, Z)
+            ("1EG4TE5M174", False),
+            ("1EG4TE5MS74", False),
+            ("1EG4TE5ML74", False),
+            ("1EG4TE5MO74", False),
+            ("1EG4TE5MOI74", False),
+            ("1EG4TE5MKB4", False),
+            ("1EG4TE5MKZ4", False),
+
+            # Position 10 is invalid
+            # Position 10 – numeric values 0 thru 9
+            ("1EG4TE5MKA4", False),
+
+            # Position 11 is invalid
+            # Position 11 – numeric values 0 thru 9
+            ("1EG4TE5MK7A", False),
+        ]
+
+        for mbi_value, expected in mbi_test_list:
+            # Create a text that contains the MBI
+            uppercase_mbi_text = f"This is a test string with MBI: {mbi_value}, expected: {expected}."
+            masked_uppercase_text = mask_mbi(uppercase_mbi_text)
+            lowercase_mbi_text = uppercase_mbi_text.lower()
+            masked_mbi_lowercase_text = mask_mbi(lowercase_mbi_text)
+            # Check if the MBI was masked
+            if expected:
+                self.assertIn('***MBI***', masked_uppercase_text)
+                self.assertIn('***MBI***', masked_mbi_lowercase_text)
+            else:
+                self.assertNotIn('***MBI***', masked_uppercase_text)
+                self.assertNotIn('***MBI***', masked_mbi_lowercase_text)
```

Comment:
We should add tests for some of the other paths that are not currently exercised in `mask_mbi`. Threw a few examples together in the example here:
```
def test_mbi_match_dict(self):
        my_dict = {'key1': '1EG4TE0MK74', 'key2': {'key4': '1EG4TE0MK74'}, 'key3': ('1EG4TE0MK74', '1EG4TE0SK74'),
                   'key5': ['1EG4TE0MK74', '1EG4TE0SK74']}
        mask_mbi(my_dict)
        print(my_dict)

        my_list = ['1EG4TE0MK74', '1EG4TE0SK74']
        mask_mbi(my_list)
        print(my_list)

        my_tuple = ('1EG4TE0MK74', '1EG4TE0SK74')
        mask_mbi(my_tuple)
        print(my_tuple)
```
Tuples and lists appear to not be working properly. It's unclear to me whether or not these need to work, because I don't have a clear picture of how these filters are used and what variety of inputs could go into the function. If the filter will only operate on strings, we can simplify the function a lot and omit these cases. But if the filter can operate on any type, we need to include tests for these cases and make adjustments to the function so that they work properly.

Reply:
added tests